server/src-lib/Hasura/RQL/DDL/Relationship.hs

```diff
@@ -151,6 +154,41 @@ createObjRelP2 (WithTable qt rd) = do
   objRelP2 qt rd
   return successMsg
 
+runCreateRemoteRelationship ::
+     (MonadTx f, CacheRM f) => CreateRemoteRelationship -> f EncJSON
+runCreateRemoteRelationship createRemoteRelationship = do
```

Comment:
we should break this into two parts:

```
createRemoteRelationshipP1
createRemoteRelationshipP2
```

The validation should go inside `createRemoteRelationshipP1` and the persist should go under `createRemoteRelationshipP2`